src/periphery/MarketLens.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// Uncomment this line to use console.log
+
+import "interfaces/IBentoBoxV1.sol";
+import "interfaces/ICauldronV2.sol";
+import {IERC20, IStrictERC20} from "BoringSolidity/interfaces/IERC20.sol";
+import "utils/CauldronLib.sol";
+
+contract MarketLens {
+    address payable public owner;
+    uint8 constant DECIMALS = 18;
+
+    constructor() {
+        owner = payable(msg.sender);
+    }
+
+    function getInterestPerYear(ICauldronV2 cauldron) external view returns (uint64 interestPerYear) {
+        (, , uint64 interestPerSecond) = cauldron.accrueInfo();
+        interestPerYear = CauldronLib.getInterestPerYearFromInterestPerSecond(interestPerSecond);
+    }
+
+    function getMaxBorrowForCauldronV2(ICauldronV2 cauldron) external view returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        maxBorrow = bentoBox.toAmount(mim, poolBalance, false);
+    }
+
+    function getMaxBorrowForCauldronV3(ICauldronV3 cauldron) external returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        uint256 mimInBentoBox = bentoBox.toAmount(mim, poolBalance, false);
+        uint256 userBorrowLimit = getUserBorrowLimit(cauldron);
+        maxBorrow = min(userBorrowLimit, mimInBentoBox);
+    }
+
+    // Why does this modify state? Want to be able to use view but cannot.
```

Comment:
Not sure why this won't let me set this as `view`.

Reply:
because `getUserBorrowLimit` is not a view

Reply:
Ah, yes, but when I add the word "view" it freaks out and gives me this error:

<img width="707" alt="Screenshot 2023-02-03 at 8 14 51 AM" src="https://user-images.githubusercontent.com/118562806/216652452-c087053c-fead-474f-be85-e6f8c8029841.png">

Reply:
Oh, easy, look at ICauldronV3.sol. borrowLimit interface is not defined correctly, just be a view. So add view to it in ICauldronV3.sol and that'll do it

Reply:
Ah, I got it. Ok, thanks!